### PR TITLE
Enable history comparison in ref-tests 

### LINKED ISSUE #1244: Fix recording new ref-tests with scrollback
When recording new ref-tests with the scrollback branch, the grid history currently is not compared at all. This leads to incorrect behavior not causing a test failure.

To make sure it's possible to write tests that also check the state of the scrollback history, not only the content of the current viewport, this should be reworked.

To keep backwards-compatibility with old ref-tests there probably needs to be a way to check what the specific history size for a ref-test should be, this can be solved by adding a `history_size` field to the grid which is already done in #TODO.

### PR DESCRIPTION
**Pull request by chrisduerr**
Previously ref-tests just ignored the scrollback history to keep
the old tests working, this would lead to new tests which rely on
scrollback history to succeeed even though they should not.

This has been fixed and it is now possible to create ref-tests with and
without scrollback history. When available the scrollback history is
compared, but the old tests still work without having to adjust them.

This fixes #1244.

This also includes #1234 and #1245.

### CODE REVIEW

File: src/grid/mod.rs [hunk 1]
```diff
@@ -52,9 +52,23 @@ impl<T> Deref for Indexed<T> {
 
 impl<T: PartialEq> ::std::cmp::PartialEq for Grid<T> {
     fn eq(&self, other: &Self) -> bool {
+        // Compare raw grid content
+        // TODO: This is pretty inefficient but only used in tests
```
Comment (jedahan) [inline src/grid/mod.rs]: is there a way to mark this with `#[cfg(test)]` or something?

Comment (PlasmaPower) [reply src/grid/mod.rs]: Yeah I think the whole impl can be marked with `#[cfg(test)]`.

Comment (chrisduerr) [reply src/grid/mod.rs]: This will not work because ref-tests are not unit tests.

Comment (jedahan) [reply src/grid/mod.rs]: Cool, it would be nice to mention why in the comments, for curious rustaceans :)

**Comment by chrisduerr** [comment]
I've reworked this PR from ground-up and it should now have a pretty efficient comparison algorithm which compares the storage are as slices in chunks.

I think this is ready for review @jwilm.

**Pull request merged**